packages/transaction-controller/src/types.ts

```diff
@@ -202,6 +203,11 @@ type TransactionMetaBase = {
    */
   id: string;
 
+  /**
+   * Whether the transaction is a smart contract interaction.
+   */
+  isSmartTransaction?: boolean;
```

Comment:
This is added but we are not setting this value for any `Transaction` right now, obviously `SmartTransactionGasFeeFlow` not in use as well. Once any client is ready to consume new API we can enable those.